### PR TITLE
Move API focused C++ samples to snippets

### PR DESCRIPTION
**Pull request by gursimarsingh**
Clean Samples #25006  
This PR removes 39 outdated C++ samples from the project, as part of an effort to keep the codebase clean and focused on current best practices.

Removed Samples:

> camshiftdemo.cpp
cloning_demo.cpp
cloning_gui.cpp
create_mask.cpp
detect_blob.cpp
dft.cpp
dis_opticalflow.cpp
distrans.cpp
ela.cpp
epipolar_lines.cpp
falsecolor.cpp
inpaint.cpp
kalman.cpp
kmeans.cpp
laplace.cpp
lsd_lines.cpp
mask_tmpl.cpp
matchmethod_orb_akaze_brisk.cpp
minarea.cpp
phase_corr.cpp
polar_transforms.cpp
segment_objects.cpp
squares.cpp
text_skewness_correction.cpp
warpPerspective_demo.cpp
watershed.cpp
imagelist_creator.cpp
imagelist_reader.cpp
imgcodecs_jpeg.cpp
intersectExample.cpp
videocapture_audio.cpp -- will be merged with videocapture_with_audio
videocapture_microphone.cpp -- will be merged with videocapture_with_audio
videocapture_openni.cpp -- will be merged with videocapture_depth
videocapture_realsense.cpp -- will be merged with videocapture_depth
videocapture_starter.cpp 
videocapture_basic.cpp 
videocapture_camera.cpp
videocapture_image_sequence.cpp
stitching.cpp -- will be merged with stitching_detailed

These changes aim to improve the repository's clarity and usability by removing examples that are no longer relevant or have been superseded by more up-to-date techniques.

### CODE REVIEW

**Comment by vpisarev** [comment]
@gursimarsingh, thanks! it looks like some of the samples are reference in documentation: https://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/106505/steps/make%20doxygen/logs/warnings%20%2818%29. Please, fix it.

**Comment by gursimarsingh** [comment]
@vpisarev Documentation build issue has been resloved

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 1]
```diff
@@ -2841,10 +2827,8 @@ the destination image will have the given size therefore the area of the boundin
 \par Reverse mapping
 
 You can get reverse mapping adding #WARP_INVERSE_MAP to `flags`
-\snippet polar_transforms.cpp InverseMap
```
Comment (mshabunin) [inline modules/imgproc/include/opencv2/imgproc.hpp:2844]: These snippets illustrated statement before them, without snippets meaning is lost. These two paragraphs must be rewritten, maybe similar snippets could be used from other sample.

Comment (gursimarsingh) [reply modules/imgproc/include/opencv2/imgproc.hpp:2844]: @mshabunin I have added the code block again, without referencing to the sample

**Comment by asmorkalov** [comment]
Hello @vpisarev @gursimarsingh @mshabunin @opencv-alalek I have alternative proposal for cleanup:
- For stitching and other samples that are part of tutorial: move the code to `samples/cpp/tutorial_code/<module>/<tutorial name>`. It's already widely used practice in OpenCV.
- For things like kmeans/svd/kalman/etc - create `snippets` or `primitives` folder and move code there. I'm strongly against of snippets removal from reference API. Also we can place python snippets there for the cases, when bindings are not obvious and use the same doxygen approach.
- Stitching detailed, calibration samples, stereo - cleanup, make more user friendly and move to apps. folder. They are more then API samples and useful themself as apps.
- Remove samples that use old C API and some legacy approaches as cascades, if they are not referenced in documentation.

**Comment by mshabunin** [comment]
> * For stitching and other samples that are part of tutorial: move the code to `samples/cpp/tutorial_code/<module>/<tutorial name>`. It's already widely used practice in OpenCV.

I think it would be more convenient to move such samples to respective modules, e.g. `modules/imgproc/samples`. Division into "samples" and "tutorial code" looks a bit artificial. And tutorials themselves can be moved to their modules as well - we already use this approach in _opencv_contrib_.

> * For things like kmeans/svd/kalman/etc - create `snippets` or `primitives` folder and move code there. I'm strongly against of snippets removal from reference API. Also we can place python snippets there for the cases, when bindings are not obvious and use the same doxygen approach.

IMO, it would be better to avoid synthetic files with code snippets, it would be better to use snippets from samples as we do now.


Also, see discussion and my comment in the issue: https://github.com/opencv/opencv/issues/25006#issuecomment-2018255279

**Review by vpisarev — APPROVED** [review]

**Comment by vpisarev** [comment]
@asmorkalov, I suggest to merge it as soon as 4.x=>5.x is merged

File: samples/cpp/detect_blob.cpp [hunk 2]
```diff
@@ -1,205 +0,0 @@
-#include <opencv2/core.hpp>
```
Comment (asmorkalov) [inline samples/cpp/detect_blob.cpp:1]: I propose to move it to snippets and mention in SimpleBlobDetector as snippet.

File: samples/cpp/dft.cpp [hunk 3]
```diff
@@ -1,86 +0,0 @@
-#include "opencv2/core.hpp"
```
Comment (asmorkalov) [inline samples/cpp/dft.cpp:1]: `cv::dft()` has code snipped, but it's embedded into header. It means that it's not built by CI. I propose to move the file to snippets, but change it's content to `convolveDFT` from core.hpp header file.

File: samples/cpp/mask_tmpl.cpp [hunk 4]
```diff
@@ -1,74 +0,0 @@
-#include "opencv2/imgproc.hpp"
```
Comment (asmorkalov) [inline samples/cpp/mask_tmpl.cpp:1]: I propose to move it to snippets and add snippet reference to matchTemplate.

File: samples/cpp/dft.cpp (hunk 3 above)
Comment (gursimarsingh) [reply samples/cpp/dft.cpp]: @asmorkalov So should we change the contents of this sample to the contents of https://github.com/opencv/opencv/blob/96a8e6d76c6351ba61a13d9f01ab7c9cd79f983f/modules/core/include/opencv2/core.hpp#L2216 

and move this sample to snippets directory and change its reference to 
`@example samples/cpp/snippets/dft.cpp` in core.hpp?

File: samples/cpp/epipolar_lines.cpp [hunk 5]
```diff
@@ -1,107 +0,0 @@
-// This file is part of OpenCV project.
```
Comment (asmorkalov) [inline samples/cpp/epipolar_lines.cpp:1]: I propose to move it to snippets and add reference for findFundamentalMat at least. There is something abstract. I propose to replace it.

File: samples/cpp/imagelist_creator.cpp [hunk 6]
```diff
@@ -1,54 +0,0 @@
-/*this creates a yaml or xml list of files from the command line args
```
Comment (asmorkalov) [inline samples/cpp/imagelist_creator.cpp:1]: I propose to presume it. Image lists are used here and there as input for other tools and samples.

File: samples/cpp/phase_corr.cpp [hunk 7]
```diff
@@ -1,46 +0,0 @@
-#include "opencv2/core.hpp"
```
Comment (asmorkalov) [inline samples/cpp/phase_corr.cpp:1]: Add ass snippet for `phaseCorrelate`

File: samples/cpp/dis_opticalflow.cpp [hunk 8]
```diff
@@ -1,64 +0,0 @@
-
```
Comment (asmorkalov) [inline samples/cpp/dis_opticalflow.cpp:1]: Convert it to snippet for DenseOpticalFlow interface and DISOpticalFlow

File: samples/cpp/cloning_demo.cpp [hunk 9]
```diff
@@ -1,251 +0,0 @@
-/*
```
Comment (asmorkalov) [inline samples/cpp/cloning_demo.cpp:1]: Propose to add it as snippet for illuminationChange, colorChange, seamlessClone, textureFlattening

File: samples/cpp/videocapture_basic.cpp [hunk 10]
```diff
@@ -1,54 +0,0 @@
-/**
-  @file videocapture_basic.cpp
```
Comment (asmorkalov) [inline samples/cpp/videocapture_basic.cpp:2]: I propose to not touch VideoCapture related samples. We have dedicated PR for it.

File: samples/cpp/opencv_version.cpp [hunk 11]
```diff
@@ -24,6 +42,31 @@ int main(int argc, const char* argv[])
     {
         std::cout << cv::getBuildInformation() << std::endl;
     }
+    else if (parser.has("env"))
+    {
+        unsigned int nThreads = std::thread::hardware_concurrency();
+        std::cout << "Number of concurrent threads supported: " << nThreads << std::endl;
```
Comment (asmorkalov) [inline samples/cpp/opencv_version.cpp:48]: I propose to throw it away.

File: samples/cpp/opencv_version.cpp [hunk 12]
```diff
@@ -24,6 +42,31 @@ int main(int argc, const char* argv[])
...
+
+        int nProcessors = cv::getNumThreads();
+        std::cout << "Number of logical processors: " << nProcessors << std::endl;
```
Comment (asmorkalov) [inline samples/cpp/opencv_version.cpp:51]: `Number of cores used by OpenCV:`

File: samples/cpp/opencv_version.cpp [hunk 13]
```diff
@@ -24,6 +42,31 @@ int main(int argc, const char* argv[])
...
+        if (cv::cuda::getCudaEnabledDeviceCount() > 0)
+        {
+            cv::cuda::DeviceInfo deviceInfo(0); // Querying the first GPU
+            std::cout << "GPU Name: " << deviceInfo.name() << std::endl;
+            std::cout << "Is integrated: " << (deviceInfo.integrated() ? "True" : "False") << std::endl;
+            std::cout << "Supports OpenGL: " << (deviceInfo.canMapHostMemory() ? "True" : "False") << std::endl;
+            std::cout << "Compute Compatibility: " << deviceInfo.majorVersion()<<"."<< deviceInfo.minorVersion() << std::endl;
+            std::cout << "Total Memory: " << deviceInfo.totalMemory() / (1024.0 * 1024.0) << " MB" << std::endl;
+
+        }
```
Comment (asmorkalov) [inline samples/cpp/opencv_version.cpp:65]: let's just loop over GPUs and print them all.

File: samples/cpp/opencv_version.cpp [hunk 14]
```diff
@@ -24,6 +42,31 @@ int main(int argc, const char* argv[])
...
+        #else
+        std::cout << "Environment: CPU" << std::endl;
+        #endif
```
Comment (asmorkalov) [inline samples/cpp/opencv_version.cpp:68]: Let's add OpenCL info too then.

**Comment by asmorkalov** [comment]
@gursimarsingh I propose to do not touch opencv_version sample in the PR and file another one. Other sample changes look good to me and I can merge the PR.

**Comment by gursimarsingh** [comment]
@asmorkalov opencv_version.cpp has been reverted and dft sample has also been improved

**Review by asmorkalov — APPROVED** [review]

**Comment by asmorkalov** [comment]
Windows warnings:
```
C:\build\precommit_windows64\5.x\opencv\samples\cpp\snippets\dft.cpp(89): warning C4838: conversion from 'double' to 'float' requires a narrowing conversion [C:\build\precommit_windows64\build\samples\cpp\example_cpp_dft.vcxproj]
C:\build\precommit_windows64\5.x\opencv\samples\cpp\snippets\dft.cpp(89): warning C4305: 'initializing': truncation from 'double' to 'float' [C:\build\precommit_windows64\build\samples\cpp\example_cpp_dft.vcxproj]
```

**Pull request merged**